src/app/calculation.cpp

```diff
@@ -85,268 +97,356 @@ QString Calculation::getError()
+            ) / 100.0F * unsealedSurfaceRunoff;
+
+            // calculate runoff 'row' for entire block patial area (FLGES +
+            // STR_FLGES) (mm/a)
+            runoff = (
+                runoffSealedSurface1 +
+                runoffSealedSurface2 +
+                runoffSealedSurface3 +
+                runoffSealedSurface4 +
+                runoffRoofs +
+                runoffPerviousRoads
+            );
 
-            // calculate runoff 'row' for entire block patial area (FLGES+STR_FLGES)
-            row = (row1 + row2 + row3 + row4 + rowd + rowuvs); // mm/a
-            ptrDA.ROW = INT_ROUND(row);
+            resultRecord.runoff = INT_ROUND(runoff);
             
-            // calculate volume 'rowvol' from runoff
-            ROWVOL = row * 3.171F * (fb + fs) / 100000.0F;     // qcm/s
+            // calculate volume 'rowvol' from runoff (qcm/s)
+            rainwaterRunoff = runoff * 3.171F * (
+                record.totalAreaBuildings +
+                record.totalAreaRoads
+            ) / 100000.0F;
             
             // calculate infiltration rate 'ri' for entire block partial area
-            ri = (ri1 + ri2 + ri3 + ri4 + rid + riuvs + riuv); // mm/a
-            ptrDA.RI = INT_ROUND(ri);
-            
-            // calculate volume 'rivol' from infiltration rate
-            RIVOL = ri * 3.171F * (fb + fs) / 100000.0F;       // qcm/s
+            // (mm/a)
+            infiltrationRate = (
+                infiltrationFromSealedSurface1 +
+                infiltrationFromSealedSurface2 +
+                infiltrationFromSealedSurface3 +
+                infiltrationFromSealedSurface4 +
+                infiltrationFromRoofs +
+                infiltrationFromPerviousRoads +
+                infiltrationFromPerviousSurfaces
+            );
+
+            resultRecord.infiltrationRate = INT_ROUND(infiltrationRate);
             
-            // calculate total system losses 'r' due to runoff and infiltration for entire block partial area
-            r = row + ri;
-            ptrDA.R = INT_ROUND(r);
+            // calculate volume 'rivol' from infiltration rate (qcm/s)
+            totalSubsurfaceFlow = infiltrationRate * 3.171F * (
+                record.totalAreaBuildings +
+                record.totalAreaRoads
+            ) / 100000.0F;
+
+            // calculate total system losses 'r' due to runoff and infiltration
+            // for entire block partial area
+            totalSystemLosses = runoff + infiltrationRate;
+
+            // set totalSystemLosses in the result record
+            resultRecord.totalSystemLosses = INT_ROUND(totalSystemLosses);
             
-            // calculate volume of system losses 'rvol'due to runoff and infiltration
-            RVOL = ROWVOL + RIVOL;
+            // calculate volume of system losses 'rvol' due to runoff and
+            // infiltration
+            totalRunoff = rainwaterRunoff + totalSubsurfaceFlow;
```

Comment:
This RVOL/totalRunoff [qcm/s] was named as 'total runoff volume'. I think this naming doesn't really match the unit. So, I would suggest the name 'totalRunoffFlow' for RVOL, 'surfaceRunoffFlow' for ROWVOL and 'infiltrationFlow' for RIVOL.

Reply:
`totalRunoffFlow = surfaceRunoffFlow + infiltrationFlow`

Reply:
Also great. Also done (see above).